hr_employee_transfer/models/employee_transfer.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+from datetime import date
+from odoo import models, fields, api, _
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+from odoo.exceptions import Warning
+
+
+class EmployeeTransfer(models.Model):
+    _name = 'employee.transfer'
+    _description = 'Employee Transfer'
+    _order = "id desc"
+
+    def _default_employee(self):
+        emp_ids = self.env['hr.employee'].search([('user_id', '=', self.env.uid)])
+        return emp_ids and emp_ids[0] or False
+
+    name = fields.Char(string='Name', help='Give a name to the Transfer', copy=False, default="/", readonly=True)
+    employee_id = fields.Many2one('hr.employee', string='Employee', required=True,
+                                  help='Select the employee you are going to transfer')
+    date = fields.Date(string='Date', default=fields.Date.today())
+    branch = fields.Many2one('transfer.company', string='Transfer Branch',  copy=False,required=True)
+
+    state = fields.Selection(
+        [('draft', 'New'), ('cancel', 'Cancelled'), ('transfer', 'Transferred'), ('done', 'Done')],
+        string='Status', readonly=True, copy=False, default='draft',
+        help=" * The 'Draft' status is used when a transfer is created and unconfirmed Transfer.\n"
+             " * The 'Transferred' status is used when the user confirm the transfer. It stays in the open status till the other branch/company receive the employee.\n"
+             " * The 'Done' status is set automatically when the employee is Joined/Received.\n"
+             " * The 'Cancelled' status is used when user cancel Transfer."
+    )
+    sequence_number = fields.Integer(string='Sequence Number', help='A unique sequence number for the Transfer',
+                                     default=1, copy=False)
+    company_id = fields.Many2one('res.company', string='Company',
+                                 related='employee_id.company_id')
+    note = fields.Text(string='Internal Notes')
+    transferred = fields.Boolean(string='Transferred', copy=False, default=False, compute='_get_transferred')
+    responsible = fields.Many2one('hr.employee', string='Responsible', default=_default_employee, readonly=True)
+
+    def _get_transferred(self):
+        print("compute")
+        if self:
+            print("self", self.branch.company_id)
+            print("self", self.env.user.company_id.id)
+            if self.branch.company_id == self.env.user.company_id.id:
+                self.transferred = True
+
+    @api.one
+    def transfer(self):
+        obj_emp = self.env['hr.employee'].browse(self.employee_id.id)
+        emp = {}
+        if not self.branch:
+            raise Warning(_(
+                'You should select the transfer branch/company.'))
+        if self.branch.company_id == self.company_id.id:
+            raise Warning(_(
+                'You cant transfer to same company.'))
+        for this in self:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused variable 'this'](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5137145)